### PR TITLE
nxos_lldp_global model

### PR DESCRIPTION
**Pull request by adharshsrivatsr**
Change in merge example

Delete example edit

### CODE REVIEW

File: models/nxos/lldp/replaced_example_01.txt [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+#lldp reinit 5
+
+- name: Replace device configuration of specific LLDP attributes with provided configuration
+  nxos_lldp:
+    config: 
+      enable: true
+      timer: 40
+    state: replaced
+
+# After state:
```
Comment (NilashishC) [inline models/nxos/lldp/replaced_example_01.txt]: Replace seems to not work as expected. Based on the task, the only lldp config lines present should be `feature lldp` and `lldp timer 40`. Other configurations should be negated.

File: models/nxos/lldp/merged_example_01.txt [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# Using merged
+# Before state:
+# -------------
+#
```
Comment (NilashishC) [inline models/nxos/lldp/merged_example_01.txt]: It's probably a good practice to specify the show command that fetches lldp config and its output, even if its empty. You can refer: https://github.com/ansible/network/pull/40/files#diff-1bb42e3752caa463e06497b7db60b669R84

Note: `% No such configuration item(s)` is what IOS-XR shows when no config items are found for a show command. NX-OS maybe different.

File: models/nxos/lldp/deleted_example_01.txt [hunk 3]
```diff
@@ -0,0 +1,17 @@
+# Using deleted
+# Before state:
+# -------------
+#
+#feature lldp
+#lldp holdtime 5
+#lldp reinit 3
+
+- name: Delete LLDP configuration (this will by default remove all lldp configuration)
```
Comment (NilashishC) [inline models/nxos/lldp/deleted_example_01.txt]: Does this stop the lldp service?

Comment (adharshsrivatsr) [reply models/nxos/lldp/deleted_example_01.txt]: Yes, if it defaulted to the previous config.

File: models/nxos/lldp/replaced_example_01.txt (hunk 1 above)
Comment (adharshsrivatsr) [reply models/nxos/lldp/replaced_example_01.txt]: Yes.

**Comment by adharshsrivatsr** [comment]
The 'tlv' attribute could be changed to 'tlv_remove' or something similar. By default, all TLVs are advertised. So we need to mention the TLVs that need not be advertised.

File: models/nxos/lldp/nxos_lldp_global.yml [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+  module: nxos_lldp_global
+  version_added: 2.9
+  short_description: Configure and manage Link Layer Discovery Protocol(LLDP) attributes on NX-OS platforms.
+  description: This module configures and manages the Link Layer Discovery Protocol(LLDP) attributes on NX-OS platforms.
+  author: Adharsh Srivats Rangarajan (@adharshsrivatsr)
+  options:
+    config:
+      description:
+        - A list of link layer discovery configurations
+      type: list
```
Comment (trishnaguha) [inline models/nxos/lldp/nxos_lldp_global.yml]: `type` should be dict.

File: models/nxos/lldp/nxos_lldp_global.yml [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+      elements: dict
```
Comment (trishnaguha) [inline models/nxos/lldp/nxos_lldp_global.yml]: No need of `elements` since type is dict

File: models/nxos/lldp/nxos_lldp_global.yml [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+      suboptions:
```
Comment (trishnaguha) [inline models/nxos/lldp/nxos_lldp_global.yml]: Add another parameter called `enabled` of type bool that enables or disables lldp.

File: models/nxos/lldp/nxos_lldp_global.yml [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+        holdtime:
+          description:
+            - Amount of time the receiving device should hold the information (in seconds)
+          type: int
+        port-id: 
+          description:
+            - This attribute defines if the interface names should be advertised in the long(0) or short(1) form.
+          type: int
+          choice: [0,1]
+          default: 0
```
Comment (trishnaguha) [inline models/nxos/lldp/nxos_lldp_global.yml]: Remove the `default` key.

File: models/nxos/lldp/nxos_lldp_global.yml [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+        reinit: 
+          description:
+            - Amount of time to delay the intialization of LLDP on any interface (in seconds)
+          type: int
+        timer:
+          description:
+            - Frequency at which LLDP updates need to be transmitted (in seconds)
+          type: int
+        tlv_remove:
```
Comment (trishnaguha) [inline models/nxos/lldp/nxos_lldp_global.yml]: Make this `tlv_select` and have the choices as suboptions for tlv_select and replace `-` by `_`.
```
        tlv_select:
          description:
            - 
          type: dict
          suboptions:
            dcbxp:
              description:
              type: bool
            port_vlan:
              description:
              type: bool
            management_address:
              description:
              type: dict
              suboptions:
                v4:
                  description:
                  type: bool
.
.
.
.
```

File: models/nxos/lldp/deleted_example_01.txt (hunk 3 above)
Comment (trishnaguha) [reply models/nxos/lldp/deleted_example_01.txt]: @adharshsrivatsr the lldp feature should be disabled when deleted.

File: models/nxos/lldp/nxos_lldp_global.yml [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+NETWORK_OS: nxos
+RESOURCE: lldp_global
+COPYRIGHT: Copyright 2019 Red Hat
+
+DOCUMENTATION: |
+  module: nxos_lldp_global
+  version_added: 2.9
+  short_description: Configure and manage Link Layer Discovery Protocol(LLDP) attributes on NX-OS platforms.
+  description: This module configures and manages the Link Layer Discovery Protocol(LLDP) attributes on NX-OS platforms.
+  author: Adharsh Srivats Rangarajan (@adharshsrivatsr)
```
Comment (trishnaguha) [inline models/nxos/lldp/nxos_lldp_global.yml]: Add `notes:` section please.

**Pull request merged**